requirements_docs.txt

```diff
@@ -1,7 +1,7 @@
+-r requirements.txt
```

Comment:
Hm... the core deps in `requirements.txt` are limited whereas the rest in `requirements_docs.txt` are pinned. This is better than before as an upgrade to NumPy will no longer break our docs CI, but is it what we want?

Reply:
I'm fine with upper limits here. Pinned requirements means a ton of dependabot PRs, which I'm ok for non-core features which make debugging doc failures a pain to track down.